lib/config/admin.ts

```diff
@@ -1,30 +1,34 @@
 import type { RenovateConfig, RepoAdminConfig } from './types';
 
-let adminConfig: RepoAdminConfig = {};
+let adminConfig: RepoAdminConfig;
 
 // TODO: once admin config work is complete, add a test to make sure this list includes all options with admin=true (#9603)
-export const repoAdminOptions = [
-  'allowCustomCrateRegistries',
-  'allowPostUpgradeCommandTemplating',
-  'allowScripts',
-  'allowedPostUpgradeCommands',
-  'customEnvVariables',
-  'dockerChildPrefix',
-  'dockerImagePrefix',
-  'dockerUser',
-  'dryRun',
-  'exposeAllEnv',
-  'privateKey',
-];
+const adminDefaults: RepoAdminConfig = {
+  allowCustomCrateRegistries: undefined,
+  allowPostUpgradeCommandTemplating: undefined,
+  allowScripts: undefined,
+  allowedPostUpgradeCommands: undefined,
+  customEnvVariables: undefined,
+  dockerChildPrefix: undefined,
+  dockerImagePrefix: undefined,
+  dockerUser: undefined,
+  dryRun: undefined,
+  exposeAllEnv: undefined,
+  privateKey: undefined,
+  localDir: '',
+  cacheDir: '',
+};
 
 export function setAdminConfig(config: RenovateConfig = {}): void {
-  adminConfig = {};
-  for (const option of repoAdminOptions) {
-    adminConfig[option] = config[option];
-    delete config[option]; // eslint-disable-line no-param-reassign
+  adminConfig = { ...adminDefaults };
```

Comment:
```suggestion
  adminConfig = {};
```

Not 100% sure of this but wouldn't it have the same result due to the for loop below it?